pkg/ca/googleca/v1/googleca.go

```diff
@@ -39,8 +39,10 @@ import (
 )
 
 type CertAuthorityService struct {
-	parent string
-	client *privateca.CertificateAuthorityClient
+	certAuthority       string
+	certAuthorityParent string
```

Comment:
I'd merge these two variables into one, `caPool`, since they represent the same resource. You can then check if `certAuthority` is set to decide whether or not to target a specific CA

Reply:
So we do need both of them later on, the full certAuthorityParent in the `GetCertificateAuthority` request and just the name of the CA `certAuthority` when actually making the request for `IssuingCertificateAuthorityId`.

i think it's more understandable to do all the validation/string splitting logic at the beginning and then have everything we need later on when actually getting CA's/making requests.

Reply:
Oh I see my confusion, it's around terminology. I thought two referred to the same resource. I would use the terms from [aip.dev](https://google.aip.dev/122) to describe the resources and IDs here:
* certAuthority -> certAuthorityID (since it represents just an ID)
* certAuthorityParent -> certAuthorityResource (since it's the full resource string)
* caPoolParent -> caPoolResource

Does that sound good? "parent" was what was confusing, since a parent of the a resource is a different resource.

Reply:
Sure, that's fine. I'd still keep "parent" personally to follow the google naming standards for a googleca, but am fine with changing it too. It's updated now.